app/hooks/activity/useActivity.tsx

```diff
@@ -7,13 +8,23 @@ export const useActivity = (userId: number) => {
   const { isLoading, data, error } = useQuery({
     queryKey,
     queryFn: async () => {
-      const result = await apis().activity.getActivities(userId);
-      return result.data;
+      try {
+        const result = await apis().activity.getActivities(userId);
+        return result.data;
+      } catch (err) {
+        if (isAccessDenied(err)) {
+          throw new Error("Access Denied");
+        } else {
+          throw err;
+        }
+      }
```

Comment:
when you copy paste something remember that there will be time when you will need to change it and all copy paste will be updated one more time.  

if you want to globally catch errors and rename them or do some behavior try out 
https://dev.to/darkmavis1980/how-to-use-axios-interceptors-to-handle-api-error-responses-2ij1 intercepting it here. 

but for this case i dont think you need this remaping. just use error object that is returned and check if isAccessDenied(error) 

eg:

 return {
    isLoading,
    activities: error ? undefined : data,
    isForbidden: isAccessDenied(error),
    error,
  };